### ISSUE TITLE
Abstract the Link component

### ISSUE THREAD

**Issue by willbreitkreutz** [USACE/groundwork#30]
Bit of a research task, need to figure out if there is a way to provide a way to bail out of <a> tags and let users provide a <Link> tag from the routing solution of their choice.

background: https://catalyst.tailwindui.com/docs#client-side-router-integration

**Comment by jbkolze** [comment]
@willbreitkreutz I'd like to get this one addressed because it's bugging me.  I know you're partial to keeping <a> tags, but I think the only good way to fix this is to provide a `<Link>` component for groundwork that defaults to `<a>` but can be overridden.

I'm thinking we could create a context wrapper with a custom link component, if desired, that would filter down to any nested <Link> components.  No wrapper, you get get an `<a>` -- within a wrapper, you get whatever you provided.  Something like

```
<LinkWrapper component={CustomLink}>
  <App>
</LinkWrapper>
```

Without the wrapper everything just works the way it does now.  Existing user code that uses `<a>` wouldn't even need to be adjusted.

Things might get a tad finicky with the way links are currently being passed as objects, but I think that can be worked around.

Thoughts?  Just wanted to run it by you before I put the elbow grease in (although it shouldn't require all that much code, honestly).

**Comment by willbreitkreutz** [comment]
@jbkolze I don't think I fully get the vision just yet but let me give it a try.  

1. you wrap the app in a context provider, give it a custom link component that belongs to the consuming app
2. in child components if you use the groundowork `<Link/>` you get either: a. `<a/>` where no wrapper/custom component is found or b. `<CustomLink />` where you wrapped it and the custom link handles nav 

I'd be interested in seeing if that works.

**Comment by jbkolze** [comment]
@willbreitkreutz Yep, that's the gist.  I'm not sensing a "please don't do this" in your response so I'll go ahead and whip up at least a draft so you can see what I have in mind.  Assuming I don't run into any unforeseen roadblocks, that is...
